x/gamm/keeper/swap_test.go

```diff
@@ -270,3 +272,47 @@ func (suite *KeeperTestSuite) TestActiveBalancerPoolSwap() {
 		}
 	}
 }
+
+// Test two pools -- one is active and should have swaps allowed,
+// while the other is inactive and should have swaps frozen.
+func (suite *KeeperTestSuite) TestInactivePoolFreezeSwaps() {
+	// Setup test
+	suite.SetupTest()
+	testCoin := sdk.NewCoin("foo", sdk.NewInt(10))
+	suite.FundAcc(suite.TestAccs[0], defaultAcctFunds)
+
+	// Setup active pool
+	activePoolId := suite.PrepareBalancerPool()
+
+	// Setup mock inactive pool
+	k := suite.App.GAMMKeeper
+	ctrl := gomock.NewController(suite.T())
+	defer ctrl.Finish()
+	inactivePool := mocks.NewMockPoolI(ctrl)
+	inactivePoolId := k.GetNextPoolNumberAndIncrement(suite.Ctx)
+	inactivePool.EXPECT().IsActive(suite.Ctx).Return(false).AnyTimes()
+	inactivePool.EXPECT().GetId().Return(inactivePoolId).AnyTimes()
```

Comment:
what's the reason for this? Wouldnt it already return the correct pool Id by defauly anyways?

Reply:
Ah this is because the `NewMockPoolI` object doesn't actually have implemented functions (like a `BalancerPool` object) -- rather all its function calls throw errors by default.

Because the swap functions use `GetId`, we have to explicitly use `Return` to ensure that it returns the correct value rather than throwing an error. Will add a comment.